docs/ai/llm-spans.md

```diff
@@ -0,0 +1,99 @@
+3. Performance concerns. Sending large amounts of data to a telemetry backend may cause performance issues for the application.
+
+By default, these configurations SHOULD NOT capture prompts and completions.
+
+## LLM Request attributes
+
+These attributes track input data and metadata for a request to an LLM. Each attribute represents a concept that is common to most LLMs.
+
+<!-- semconv ai(tag=llm-request) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `llm.vendor` | string | The name of the LLM foundation model vendor, if applicable. If not using a vendor-supplied model, this field is left blank. | `openai` | Recommended |
+| `llm.request.model` | string | The name of the LLM a request is being made to. If the LLM is supplied by a vendor, then the value must be the exact name of the model requested. If the LLM is a fine-tuned custom model, the value SHOULD have a more specific name than the base model that's been fine-tuned. | `gpt-4` | Required |
+| `llm.request.max_tokens` | int | The maximum number of tokens the LLM generates for a request. | `100` | Recommended |
+| `llm.temperature` | float | The temperature setting for the LLM request. | `0.0` | Recommended |
+| `llm.top_p` | float | The top_p sampling setting for the LLM request. | `1.0` | Recommended |
+| `llm.stream` | bool | Whether the LLM responds with a stream. | `false` | Recommended |
+| `llm.stop_sequences` | array | Array of strings the LLM uses as a stop sequence. | `["stop1"]` | Recommended |
+
+`llm.model` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `gpt-4` | GPT-4 |
+| `gpt-4-32k` | GPT-4 with 32k context window |
+| `gpt-3.5-turbo` | GPT-3.5-turbo |
+| `gpt-3.5-turbo-16k` | GPT-3.5-turbo with 16k context window|
+| `claude-instant-1` | Claude Instant (latest version) |
+| `claude-2` | Claude 2 (latest version) |
+| `other-llm` | Any LLM not listed in this table. Use for any fine-tuned version of a model. |
+<!-- endsemconv -->
+
+## LLM Response attributes
+
+These attributes track output data and metadata for a response from an LLM. Each attribute represents a concept that is common to most LLMs.
+
+<!-- semconv ai(tag=llm-response) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `llm.response.id` | string | The unique identifier for the completion. | `chatcmpl-123` | Recommended |
+| `llm.response.model` | string | The name of the LLM a response is being made to. If the LLM is supplied by a vendor, then the value must be the exact name of the model actually used. If the LLM is a fine-tuned custom model, the value SHOULD have a more specific name than the base model that's been fine-tuned. | `gpt-4-0613` | Required |
```

Comment:
Do we expect llm.request.model and llm.response.model to be different? The request and response are all recorded on one span, so these would be redundant?

Reply:
Yes, for example in OpenAI you ask for `gpt-4` and then get a specific version like `gpt-4-0613` (I've also seen this in Ahthropic, Replicate, and others)